pkg/steps/multi_stage.go

```diff
@@ -472,6 +473,14 @@ func (s *multiStageTestStep) generatePods(steps []api.LiteralTestStep, env []cor
 			errs = append(errs, err)
 			continue
 		}
+		allResources := &resources
+		if !resources.Limits.Name(api.ShmResource, resource.BinarySI).IsZero() {
```

Comment:
I'm not sure I entirely follow - I think we should check requests here and not limits?

Reply:
Since we have to set them both to pass k8s validations we could check for the presence of either here, but I agree that checking `Requests` makes more sense and will update that.